javaslang/generator/Generator.scala

```diff
@@ -1136,6 +1136,28 @@ def generateMainClasses(): Unit = {
              * @throws NullPointerException if {@code f} is null
              */
             ${if (i == 0) xs"""
+              public <U> U apply($functionType<? extends U> f) {
+                  $Objects.requireNonNull(f, "f is null");
+                  return f.get();
+              }
+            """ else xs"""
+              public <U> U apply($functionType<$paramTypes, ? extends U> f) {
+                  $Objects.requireNonNull(f, "f is null");
+                  return f.apply($params);
+              }
+            """}
+
+            /**
+             * Transforms this tuple to an object of type U.
+             *
+             * @deprecated Use {@link #apply($functionType)} instead, will be removed in 3.0.0
+             * @param f Transformation which creates a new object of type U based on this tuple's contents.
+             * @param <U> type of the transformation result
+             * @return An object of type U
+             * @throws NullPointerException if {@code f} is null
+             */
+            @Deprecated(/* Use apply instead, will be removed in 3.0.0 */)
+            ${if (i == 0) xs"""
               public <U> U transform($functionType<? extends U> f) {
                   $Objects.requireNonNull(f, "f is null");
```

Comment:
Great, thank you!

I think it does not really matter (because it is generated code) but could you delegate to `apply`?

```scala
public <U> U transform($functionType<? extends U> f) {
    return apply(f);
}
```